File: part3/nodejs/httpServer.py
import sys
sys.path.append('gen-py')

from hello import helloSvc
from thrift.protocol import TJSONProtocol
from thrift.server import THttpServer

class HelloSvcHandler:
  def getMessage(self, name):
    print "Called with " + name
    return "hello " + name

processor = helloSvc.Processor(HelloSvcHandler())
protoFactory = TJSONProtocol.TJSONProtocolFactory()
port = 9090
server = THttpServer.THttpServer(processor, ("localhost", port), protoFactory)
print "Python server running on port " + str(port)
server.serve()

File: part3/js/httpServer.js
var thrift = require('thrift');                 	
var helloSvc = require('./gen-nodejs/helloSvc');		

//ServiceHandler: Implement the hello service 
var helloHandler = {
  getMessage: function (name, result) {
    console.log("Received: " + name);
    result(null, "Hello " + name);
  }
};

//ServiceOptions: The I/O stack for the service
var helloSvcOpt = {                       		
    handler: helloHandler,                      	
    processor: helloSvc,                         	
    protocol: thrift.TJSONProtocol,                 
    transport: thrift.TBufferedTransport 		
};                                  

//ServerOptions: Define server features
var serverOpt = {                          	
   services: {                         
      "/hello": helloSvcOpt                 
   }                               
}                                   

//Create and start the web server 
var port = 9090;                            		
thrift.createWebServer(serverOpt).listen(port);                        		
console.log("Http/Thrift Server running on port: " + port);

